### PR TITLE
GO-332 Add proper copy-paste plain text rendering

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: util/strutil/str.go [hunk 1]
```diff
@@ -1,9 +1,45 @@
...
+
+func JoinWithTrailingEnd(elems []string, sep string) string {
+	switch len(elems) {
+	case 0:
+		return ""
+	case 1:
+		return elems[0] + sep
+	}
+
+	var n int
```
Comment (deff7) [inline util/strutil/str.go:24]: Is this preallocation so necessary? strings.Builder should work pretty well

File: util/strutil/str.go [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
 func CapitalizeFirstLetter(str string) string {
 	var runes = []rune(str)
 	runes = append([]rune{unicode.ToUpper(runes[0])}, runes[1:]...)
 	return string(runes)
 }
+
+func JoinWithTrailingEnd(elems []string, sep string) string {
+	switch len(elems) {
+	case 0:
+		return ""
```
Comment (deff7) [inline util/strutil/str.go:19]: `return sep`? Or is it a corner case?

File: core/block/editor/clipboard/clipboard.go [hunk 3]
```diff
@@ -564,32 +565,54 @@ func (cb *clipboard) addRelationLinksToDataview(d *model.BlockContentDataview) (
...
+	texts = renderChildren(s, texts, childrenIds, level, 0)
+	return texts, numberedCount
+}
+
+func renderChildren(s *state.State, texts []string, childrenIds []string, level int, numberedCount int) []string {
+	var oldNumberedCount int
+	for _, id := range childrenIds {
+		oldNumberedCount = numberedCount
+		texts, numberedCount = renderBlock(s, texts, id, level+1, numberedCount)
+		if oldNumberedCount == numberedCount {
```
Comment (deff7) [inline core/block/editor/clipboard/clipboard.go:590]: Is it for splitting numbered list for separate chunks? Like

```
# One
# Two
Text
# One
```
renders to
```
1. One
2. Two
Text
1. One
```

Comment (fat-fellow) [reply core/block/editor/clipboard/clipboard.go:590]: yep

File: util/strutil/str.go (hunk 2 above)
Comment (fat-fellow) [reply util/strutil/str.go:19]: yeah, it is

File: util/strutil/str.go (hunk 1 above)
Comment (fat-fellow) [reply util/strutil/str.go:24]: i am simply copied and modified default strings.Join, tried to do less changes, gonna add a comment about it

**Review by deff7 — APPROVED** [review]

**Pull request merged**